src/rules/deprecation.ts

```diff
@@ -13,84 +13,182 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-import { TSESTree } from '@typescript-eslint/experimental-utils';
+import {
+  ESLintUtils,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
 import { ParserServices } from '@typescript-eslint/parser';
-import { Rule } from 'eslint';
-import { getParent } from 'eslint-plugin-sonarjs/lib/utils/nodes';
-import * as estree from 'estree';
+import { isReassignmentTarget } from 'tsutils';
 import * as ts from 'typescript';
 
-export default {
-  create(context: Rule.RuleContext) {
+const createRule = ESLintUtils.RuleCreator(
+  () => 'https://github.com/gund/eslint-plugin-deprecation',
+);
+
+export type Options = unknown[];
+export type MessageIds = 'deprecated';
+
+export default createRule<Options, MessageIds>({
+  name: 'deprecation',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Do not use deprecated APIs.',
+      category: 'Best Practices',
+      recommended: 'warn',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      deprecated: `'{{name}}' is deprecated. {{reason}}`,
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
     const services = context.parserServices;
     if (!isRequiredParserServices(services)) {
+      console.warn(
+        'TS not configured correctly for rule deprecation/deprecation',
+      );
       return {};
     }
```

Comment:
```suggestion
    const services = ESLintUtils.getParserServices(context);
```

Reply:
But will this throw an error or just return `undefined`?

Reply:
it will throw an error, which is the same as all rules that require type information in `@typescript-eslint`.